Changelog 24.7.0 (#5418)
conda/conda-build: CHANGELOG.md

[//]: # (current developments)

## 24.7.0 (2024-07-18)

### Enhancements

* Skip generating `repodata.json.bz2` for local index; generate `repodata.json`
  only; require `conda-package-handling >=2.2.0` matching conda. (#5231)
* Add new include/exclude sections for glob expressions in multi-output `outputs/files`. (#4196 via #5216)

> * Increased performance by using pickle instead of deepcopy. (#5281)

Reviewer: ```suggestion
* Increase performance by using `pickle` instead of `copy.deepcopy`. (#5281)
```

* Report fully rendered recipe to stdout before the build process starts. (#3798 via #5344)
* Validate `run_constrained` dependencies to prevent faulty specs reaching final repodata. (#5047 via #5359)

> * PIP_* env variables are set when building outputs in multi-output recipes. (#3993 via #5368)

Reviewer: ```suggestion
* `PIP_*` env variables are set when building outputs in multi-output recipes. (#3993 via #5368)
```

> * Reduced performance overheads of logging. (#5384)

Reviewer: ```suggestion
* Reduce performance overhead of logging. (#5384)
```

  * `conda_build.conda_interface.url_path` (use `conda.utils.url_path` instead). (#5333)
  * `conda_build.conda_interface.VersionOrder` (use `conda.models.version.VersionOrder` instead). (#5333)
  * `conda_build.conda_interface.walk_prefix` (use `conda.misc.walk_prefix` instead). (#5333)
  * `conda_build.conda_interface.win_path_to_unix` (use `conda.common.path.win_path_to_unix` instead). (#5333)
  * `conda_build.index.channel_data`; `conda_build.index.get_build_index` return value for `channel_data` is now always `None`. (#5333)
  * `conda_build.metadata.check_circular_dependencies` (use `conda_build.metadata._check_circular_dependencies` instead). (#5406)
  * `conda_build.metadata.toposort` (use `conda_build.metadata.toposort_outputs` instead). (#5406)
  * `conda_build.utils._convert_lists_to_sets` (use `frozendict.deepfreeze` instead). (#5333)
  * `conda_build.utils.HashableDict` (use `frozendict.deepfreeze` instead). (#5333)
  * `conda_build.utils.represent_hashabledict` (use `frozendict.deepfreeze` instead). (#5333)
  * `conda_build.utils.rm_rf(config)`. (#5333)
  * `conda_build.variants.get_vars(loop_only)`. (#5333)

### Other

> * Long `conda info` error messages for common exceptions. (#5264)

Reviewer: Long? Log?

Author: ```suggestion
* Log `conda info` error messages for common exceptions. (#5264)
```
🪵

Reviewer: ```suggestion
* Do not generate conda error reports for common exceptions. (#5264)
```

Author: Oops, good catch! I cut + pasted the issue title, which is what got fixed (the error reports were too long)

* Add `CondaBuildUserError` exception to replace `sys.exit` calls. (#5353)

> * Updated the CI to download the MacOSX 10.15 SDK.

Reviewer: ```suggestion
* Updated the CI to download the MacOSX 10.15 SDK. (#5387)
```
